enterprise/course_catalog_api.py

```diff
@@ -22,50 +24,93 @@
     """
 
-    pass
+    def __init__(self, user):
+        """
+        Create an Enrollment API client, authenticated with the API token from Django settings.
 
+        This method retrieves an authenticated API client that can be used
+        to access the course catalog API. It raises an exception to be caught at
+        a higher level if the package doesn't have OpenEdX resources available.
+        """
+        if CatalogIntegration is None:
+            raise NotConnectedToOpenEdX(
+                _('To get a CatalogIntegration object, this package must be installed in an OpenEdX environment.')
+            )
+        if get_edx_api_data is None:
+            raise NotConnectedToOpenEdX(
+                _('To parse a catalog API response, this package must be installed in an OpenEdX environment.')
+            )
+        if course_discovery_api_client is None:
+            raise NotConnectedToOpenEdX(
+                _('To get a catalog API client, this package must be installed in an OpenEdX environment.')
+            )
 
-def get_catalog_api_client(user):
-    """
-    Retrieve a course catalog API client.
+        self.user = user
+        self.client = course_discovery_api_client(user)
 
-    This method retrieves an authenticated API client that can be used
-    to access the course catalog API. It raises an exception to be caught at
-    a higher level if the package doesn't have OpenEdX resources available.
-    """
-    if course_discovery_api_client is None:
-        raise NotConnectedToOpenEdX(
-            _('To get a catalog API client, this package must be installed in an OpenEdX environment.')
-        )
-    else:
-        return course_discovery_api_client(user)
+    def get_all_catalogs(self):
+        """
+        Return a list of all course catalogs, including name and ID.
+        """
+        return self._load_data('catalogs')
 
+    def get_course_run(self, course_run_id):
+        """
+        Return course_run data, including name, ID and seats.
+        """
+        return self._load_data('course_runs', resource_id=course_run_id)
 
-def get_all_catalogs(user):
-    """
-    Return a list of all course catalogs, including name and ID.
-    """
-    client = get_catalog_api_client(user)
-    if CatalogIntegration is None:
-        raise NotConnectedToOpenEdX(
-            _('To get a CatalogIntegration object, this package must be installed in an OpenEdX environment.')
-        )
-    if get_edx_api_data is None:
-        raise NotConnectedToOpenEdX(
-            _('To parse a catalog API response, this package must be installed in an OpenEdX environment.')
+    def get_program_by_title(self, program_title):
+        """
+        Return single program by name, or None if not found.
+        """
+        all_programs = self._load_data('programs')
+        try:
+            return next(program for program in all_programs if program.get('title') == program_title)
```

Comment:
Should we return a single program in a scenario where multiple programs match the same title? Or should we raise an exception to complain about the ambiguity?

Better yet, we might want to yield each matching program from here, and then raise later on if either none of the programs are eligible for enrollment, or if multiple programs are eligible for enrollment.